plugins/services/src/js/components/forms/EnvironmentFormSection.js

```diff
@@ -0,0 +1,213 @@
+          </FormGroup>
+          <FormGroup
+            className="column-3"
+            required={false}
+            showError={Boolean(errors.env[key])}>
+            {valueLabel}
+            <FieldInput
+              name={`env.${key}.value`}
+              type="text"
+              value={env.value}/>
+            <FieldError>{errors.env[key]}</FieldError>
+          </FormGroup>
+          <div className="form-group flex flex-item-align-end column-4">
+            <a className="column-3 button button-primary-link button-flush"
+              data-path="env"
+              data-value={key}
+              onClick={this.handleOnRemoveItem}>
+              Delete
+            </a>
+          </div>
+        </div>
+    ); });
+  }
+
+  getLabelsLines(data) {
+    const {errors} = this.props;
+    return data.map((label, key) => {
+      let keyLabel = null;
+      let valueLabel = null;
+      if (key === 0) {
+        keyLabel = <FieldLabel>Key</FieldLabel>;
+        valueLabel = <FieldLabel>Value</FieldLabel>;
+      }
+      return (
+        <div key={key} className="flex row">
+          <FormGroup
+            className="column-3"
+            required={false}
+            showError={Boolean(errors.labels[key])}>
+            {keyLabel}
+            <FieldInput
+              name={`labels.${key}.key`}
+              type="text"
+              value={label.key}/>
+            <span className="emphasis form-colon">:</span>
+            <FieldError>{errors.labels[key]}</FieldError>
+          </FormGroup>
+          <FormGroup
+            className="column-3"
+            required={false}
+            showError={Boolean(errors.labels[key])}>
+            {valueLabel}
+            <FieldInput
+              name={`labels.${key}.value`}
+              type="text"
+              value={label.value}/>
+            <FieldError>{errors.labels[key]}</FieldError>
+          </FormGroup>
+          <div className="form-group flex flex-item-align-end column-4">
+            <a className="column-3 button button-primary-link button-flush"
+              data-path="labels"
+              data-value={key}
+              onClick={this.handleOnRemoveItem}>
+              Delete
+            </a>
+          </div>
+        </div>
+      );
+    });
+  }
+
+  render() {
+    let {data} = this.props;
+
+    return (
+      <div className="form flush-bottom">
+        <div className="form-row-element">
+          <h2 className="form-header flush-top short-bottom">
+            Environment Variables
+          </h2>
+          <p>
+            Set up environment variables for each task your service launches.
+          </p>
+        </div>
+        {this.getEnvironmentLines(data.env)}
+        <div>
+          <a className="button button-primary-link button-flush"
+            data-value={data.env.length}
+            data-path="env"
+            onClick={this.handleOnAddItem}>
+            + Add Environment Variable
+          </a>
```

Comment:
I would like for us not to use the `dataset` for this add button, similar to my comment about the delete button.